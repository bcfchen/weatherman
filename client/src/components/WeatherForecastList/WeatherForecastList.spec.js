import React from "react";
import { shallow } from "enzyme";
import WeatherForecastList from "./WeatherForecastList";
import DailyForecast from "../../models/DailyForecast";

describe("<WeatherForecastList/>", () => {
    let component = null,
        forecastProps = {
            Temperature: {
                Maximum: { Value: 15 },
                Minimum: { Value: 9 }
            },
            Day: { IconPhrase: "Sunny" },
            Date: "20180915"
        },
        defaultProps = {
            weatherForecasts: [new DailyForecast(forecastProps)]
        };
    let render = (props = defaultProps) => {
        component = shallow(<WeatherForecastList {...props} />);
    };
    beforeEach(() => render());

    describe("on page load", () => {
        it("should render correctly", () => {
            expect(component).toMatchSnapshot();
        });

        it("should contain the correct number of WeatherForecastItem components", () => {
            expect(component.find("WeatherForecastItem")).toHaveLength(1);
        });
    });
});