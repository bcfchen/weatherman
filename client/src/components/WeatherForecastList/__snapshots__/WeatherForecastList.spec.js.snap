// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeatherForecastList/> on page load should render correctly 1`] = `
<WithStyles(Grid)
  className="weather-forecast-items-container"
  container={true}
>
  <WithStyles(Grid)
    item={true}
    key="0"
    xs={true}
  >
    <WeatherForecastItem
      forecastItem={
        DailyForecast {
          "dayOfWeek": "Sat",
          "high": 15,
          "low": 9,
          "text": "Sunny",
        }
      }
    />
  </WithStyles(Grid)>
</WithStyles(Grid)>
`;
